File: Python/refer-offer/gu_piao_de_zui_da_li_run_lcof.py
from typing import *

def maxProfit(prices: List[int]) -> int:
    length = len(prices)
    if length < 2:
        return 0
    buy, sell = -prices[0], 0
    for i in range(1, length):
        buy = max(buy, -prices[i])
        sell = max(sell, buy + prices[i])
    
    return sell

if __name__ == "__main__":
    print(maxProfit([7,1,5,3,6,4]))

File: Java/src/com/dbc/algorithm/ReferOffer/GuPiaoDeZuiDaLiRunLcof.java
package com.dbc.algorithm.ReferOffer;

public class GuPiaoDeZuiDaLiRunLcof {
    public int maxProfit(int[] prices) {
        if (prices.length < 2) return 0;
        int sell = 0, buy = -prices[0];
        for (int i = 1; i < prices.length; i++) {
            buy = Math.max(buy, -prices[i]);
            sell = Math.max(sell, buy + prices[i]);
        }
        return sell;
    }
}
